In [60]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
import glob
import re
import time

In [48]:
lista_link_ADE = list()

### ADE's pesquisados com os filtros: "inexistente de fato", data do ato entre 01/01/2017 e 15/04/2021:

In [49]:
# Obtém o html da página web que possui os links de todos os ADE's que precisamos extrair:
response = requests.get('http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=tipoAtoFacet%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=Ato+Declarat%C3%B3rio+Executivo&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=%22inexistente+de+fato%22&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=1&optOrdem=relevancia&p=1')
soup = BeautifulSoup(response.text, 'html.parser')

In [50]:
# Obtém os elementos <tr> que possuem os links para os ADE:
lista_tag_tr = soup.find_all("tr", {"class": "linhaResultados"})

# Obtém os links dos ADE:
for tr in lista_tag_tr:
    lista_link_ADE.append(tr.find("a").get('href'))
    
print('Quantidade de ADE: ', len(lista_link_ADE))

Quantidade de ADE:  84


Como a paginação retornada pelo site é de até 100 itens por página, e foram retornados apenas 84, então não precisamos fazer paginação.

### ADE's pesquisados com os filtros: "inexistência de fato", data do ato entre 01/01/2017 e 15/04/2021:

In [51]:
# Via acesso manual, verificamos que há 4 páginas. Então, será necessário percorrê-las.
for pagina in range(4):
    # Obtém o html da página web que possui os links dos ADE's que precisamos extrair:
    url = 'http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=tipoAtoFacet%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=Ato+Declarat%C3%B3rio+Executivo&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=%22inexist%C3%AAncia+de+fato%22&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=' + str(pagina+1) +  '&optOrdem=relevancia'
    print('Acessando: ', url)
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')
   
    # Obtém os elementos <tr> que possuem os links para os ADE:
    lista_tag_tr = soup.find_all("tr", {"class": "linhaResultados"})

    print('Resultados na página: ', len(lista_tag_tr))
    
    # Obtém os links dos ADE:
    for tr in lista_tag_tr:
        lista_link_ADE.append(tr.find("a").get('href'))

Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=tipoAtoFacet%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=Ato+Declarat%C3%B3rio+Executivo&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=%22inexist%C3%AAncia+de+fato%22&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=1&optOrdem=relevancia
Resultados na página:  100
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=tipoAtoFacet%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=Ato+Declarat%C3%B3rio+Executivo&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=%22inexist%C3%AAncia+de+fato%22&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=2&optOrdem=relevancia
Resultados na página:  100
Acessando:  http://normas.receita.faze

O resultado acima (100+100+100+58) bate com o que está escrito na página de pesquisa: "Total de atos localizados: 358 (100 por página)".

### ADE's pesquisados com os filtros: DECLAR* AND BAIXA* AND OFÍCIO, data do ato entre 01/01/2017 e 15/04/2021:

In [52]:
# Via acesso manual, verificamos que há 7 páginas. Então, será necessário percorrê-las.
for pagina in range(7):
    # Obtém o html da página web que possui os links dos ADE's que precisamos extrair:
    url = 'http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=DECLAR*+AND+BAIXA*+AND+OF%C3%8DCIO&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=' + str(pagina+1) + '&optOrdem=relevancia'
    print('Acessando: ', url)
    response = requests.get(url)
    soup = BeautifulSoup(response.text, 'html.parser')
   
    # Obtém os elementos <tr> que possuem os links para os ADE:
    lista_tag_tr = soup.find_all("tr", {"class": "linhaResultados"})

    print('Resultados na página: ', len(lista_tag_tr))
    
    # Obtém os links dos ADE:
    for tr in lista_tag_tr:
        lista_link_ADE.append(tr.find("a").get('href'))

Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=DECLAR*+AND+BAIXA*+AND+OF%C3%8DCIO&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=1&optOrdem=relevancia
Resultados na página:  100
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=%2C&orgaosSelecionados=&tiposAtosSelecionados=&lblTiposAtosSelecionados=&ordemColuna=&ordemDirecao=&tipoConsulta=formulario&tipoAtoFacet=&siglaOrgaoFacet=&anoAtoFacet=&termoBusca=DECLAR*+AND+BAIXA*+AND+OF%C3%8DCIO&numero_ato=&tipoData=1&dt_inicio=01%2F01%2F2017&dt_fim=15%2F04%2F2021&ano_ato=&p=2&optOrdem=relevancia
Resultados na página:  100
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/consulta.action?facetsExistentes=%2C&orgaosSelecionado

O resultado acima (100+100+100+100+100+100+69) bate com o que está escrito na página de pesquisa: "Total de atos localizados: 699 (100 por página)".

### Temos então a seguinte quantidade de ADE para extrair os CNPJ baixados por inexistência de fato:

In [53]:
len(lista_link_ADE)

1141

In [54]:
lista_link_ADE

['link.action?visao=anotado&idAto=98462',
 'link.action?visao=anotado&idAto=97332',
 'link.action?visao=anotado&idAto=96744',
 'link.action?visao=anotado&idAto=96270',
 'link.action?visao=anotado&idAto=96269',
 'link.action?visao=anotado&idAto=96268',
 'link.action?visao=anotado&idAto=92474',
 'link.action?visao=anotado&idAto=91905',
 'link.action?visao=anotado&idAto=85498',
 'link.action?visao=anotado&idAto=82427',
 'link.action?visao=anotado&idAto=81622',
 'link.action?visao=anotado&idAto=81621',
 'link.action?visao=anotado&idAto=80267',
 'link.action?visao=anotado&idAto=79546',
 'link.action?visao=anotado&idAto=79519',
 'link.action?visao=anotado&idAto=116161',
 'link.action?visao=anotado&idAto=116014',
 'link.action?visao=anotado&idAto=115616',
 'link.action?visao=anotado&idAto=115175',
 'link.action?visao=anotado&idAto=115149',
 'link.action?visao=anotado&idAto=115057',
 'link.action?visao=anotado&idAto=115012',
 'link.action?visao=anotado&idAto=115013',
 'link.action?visao=anotad

Vamos fazer ajuste dos links:

In [55]:
nova_lista_link_ADE = list()
for link in lista_link_ADE:
    nova_lista_link_ADE.append('http://normas.receita.fazenda.gov.br/sijut2consulta/' + link.replace("anotado", "compilado"))

In [56]:
# Removendo duplicados (Converve para dicionário e depois para lista de volta):
nova_lista_link_ADE = list( dict.fromkeys(nova_lista_link_ADE) )
len(nova_lista_link_ADE)

786

Após ajustes, temos 786 atos para pesquisar e extrair os CNPJs.

### Vamos agora acessar cada página e extrair os CNPJs:

In [65]:
pattern = re.compile('\D(\d?\d\D?\d{3}\D?\d{3}\D?\d{4}\s?\D?\s?\d{2})')
empresas_baixadas_df = pd.DataFrame(columns=['CNPJ', 'ADE', 'Link'])
# contador = 0
for link in nova_lista_link_ADE:
    print('Acessando: ', link)
    
    # Para contornar problemas de conexão:
    for tentativa in range(3):
        try:
            response = requests.get(link, headers={"User-Agent" : "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36"})
        except Exception as e:
            print('Erro de conexão (', tentativa, '): ', e)
    
    if not response:
        print('  > LINK PULADO!!!!')
        continue
    
    soup = BeautifulSoup(response.text, 'html.parser')
    
    # Obtém o texto do ADE e faz algumas verificações:
    texto = soup.find("body").text.upper()
    
    # Verifica se o ADE possui alguma variação do texto INEXISTE DE FATO:
    if not (re.search('INEXIST.{1,6} DE FATO', texto)):
        # Como não encontrou o texto, verifica se há menção aos normativos que regulam a baixa de ofício por inexistência de fato
        if (re.search('(INSTRUÇÃO NORMATIVA|IN).+(1.?863.+?2.?018|1.?634.+?2.?016)', texto) == None) or (re.search('ART.{1,6}(29|31)', texto) == None):
            # Verifica se é um ato que não tem mais efeito, ou se é de baixa de CPF:
            m = re.search('(ANULADO|REVOGAGO|SUSPENSO|SEM EFEITO|CPF)', texto)
            print('  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016!', m.groups() if m else '')
            continue
    
    # Verifica se o ato efetivamente 'declara'
    if re.search("(DECLAR.{1,5})", texto) == None:
        print('  > Não encontrou o termo DECLARA!')
        continue
    
    cnpjs_encontrados = pattern.findall(texto)
    
    qte_cnpjs_encontrados = len(cnpjs_encontrados)
    
    if qte_cnpjs_encontrados == 0:
        m = re.search('(ANULAD.|REVOGAD.|SUSPENS.|SEM EFEITO|CPF)', texto)
        print('  > Nenhum CNPJ encontrado!', m.groups() if m else '')
        continue
            
    print('  > CNPJs encontrados: ', qte_cnpjs_encontrados)
    
    # Obtém o título do ADE
    ADE = soup.find("div", {"class": "tituloAto"}).text
    ADE = re.sub(r'[\t\n]', '', ADE)
    ADE = re.sub(r'\s+', ' ', ADE)
    
    temp_df = pd.DataFrame(columns=['CNPJ', 'ADE', 'Link'])
    temp_df['CNPJ'] = cnpjs_encontrados
    temp_df['ADE'] = ADE
    temp_df['Link'] = link

    empresas_baixadas_df = pd.concat([empresas_baixadas_df, temp_df], ignore_index=True)    
    
empresas_baixadas_df

Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98462
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=97332
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96744
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96270
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96269
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96268
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=92474
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilad

  > CNPJs encontrados:  7
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88925
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88924
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88923
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=87677
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86938
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86663
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=85902
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/

  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=90659
  > CNPJs encontrados:  22
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=90034
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=89346
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=89345
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=89344
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88840
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88839
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta

  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82109
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82108
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82107
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82106
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82105
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81690
  > CNPJs encontrados:  3
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81506
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/

  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98247
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98246
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98245
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98244
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98243
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98242
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98241
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/

  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88659
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88658
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88535
  > Nenhum CNPJ encontrado! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88534
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88203
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88202
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=88201
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consul

  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! ('ANULADO',)
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81982
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81961
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81927
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81926
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81925
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81924
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81923
  > CNPJs encontrados:  1
Aces

  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=97262
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96602
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96310
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96309
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96031
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=95726
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=95603
  > CNPJs encontrados:  1
Acessando:  http

  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=87116
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86634
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86378
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86377
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86328
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86172
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=

  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=104558
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=103453
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=103452
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=102867
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=102781
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=102592
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=102449
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2co

  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=85671
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=85509
  > CNPJs encontrados:  3
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=84867
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=84543
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=84226
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=84176
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=84164
  > CNPJs encontrados:  2
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/

  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=105647
  > CNPJs encontrados:  1
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=101496
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=101335
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=99709
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98777
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao

  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=111882
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=111610
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=110465
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=108669
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=108608
  > Nenhum CNPJ encontrado! ('SUSPENSÃ',)
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta

  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! ('CPF',)
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=87209
Erro de conexão ( 0 ):  ('Connection broken: IncompleteRead(7582 bytes read, 602 more expected)', IncompleteRead(7582 bytes read, 602 more expected))
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! ('CPF',)
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86792
  > Não encontrou inexistente de fato e nem IN 1.863/2018 ou 1.634/2016! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=86551
  > Nenhum CNPJ encontrado! 
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=83554
  > Nenhum CNPJ encontrado! ('SUSPENSÃ',)
Acessando:  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82568
  > Não encontrou ine

CNPJ                                                ADE  \
0     13.255.232/0001-32  Ato Declaratório Executivo DRF/MCR nº 1, de 29...   
1     71.064.976/0001-76  Ato Declaratório Executivo DRF/MCR nº 1, de 29...   
2     05.200.212/0001-36  Ato Declaratório Executivo DRF/MCR nº 39, de 1...   
3     15.545.768/0001-18  Ato Declaratório Executivo DRF/MCR nº 37, de 2...   
4     07.561.035/0001-49  Ato Declaratório Executivo DRF/MCR nº 31, de 2...   
...                  ...                                                ...   
1275  10.530.609/0001-26  Ato Declaratório Executivo DRF/AJU nº 6, de 15...   
1276  05.705.786/0001-66  Ato Declaratório Executivo DRF/AJU nº 1, de 27...   
1277  39.395.132/0001-03  Ato Declaratório Executivo DRF/VIT nº 129, de ...   
1278  07.857.451/0001-99  Ato Declaratório Executivo DRF/PAL nº 17, de 2...   
1279  07.857.451/0001-99  Ato Declaratório Executivo DRF/PAL nº 17, de 2...   

                                                   Link  
0     http://normas.receita.fazenda.gov.br/sijut2con...  
1     http://normas.receita.fazenda.gov.br/sijut2con...  
2     http://normas.receita.fazenda.gov.br/sijut2con...  
3     http://normas.receita.fazenda.gov.br/sijut2con...  
4     http://normas.receita.fazenda.gov.br/sijut2con...  
...                                                 ...  
1275  http://normas.receita.fazenda.gov.br/sijut2con...  
1276  http://normas.receita.fazenda.gov.br/sijut2con...  
1277  http://normas.receita.fazenda.gov.br/sijut2con...  
1278  http://normas.receita.fazenda.gov.br/sijut2con...  
1279  http://normas.receita.fazenda.gov.br/sijut2con...  

[1280 rows x 3 columns]

In [66]:
# Salvando checkpoint:
empresas_baixadas_df.to_csv(r'C:\DATASETS\Empresas Baixadas\empresas_baixadas_extraido_site_normas.csv', index = False)

In [67]:
(empresas_baixadas_df.isnull().sum() / empresas_baixadas_df.shape[0] * 100).sort_values(ascending=False)

Link    0.0
ADE     0.0
CNPJ    0.0
dtype: float64

Nenhum nulo.

In [68]:
empresas_baixadas_df.CNPJ.str.len().value_counts()

18    1252
17      10
15       9
14       5
16       4
Name: CNPJ, dtype: int64

In [79]:
# Vamos tirar a formatação e remover os duplicados:
empresas_baixadas_df.CNPJ = empresas_baixadas_df.CNPJ.str.replace('[\.\-\/\\n]', '', regex=True)
empresas_baixadas_df = empresas_baixadas_df.drop_duplicates(subset=['CNPJ'])
print(empresas_baixadas_df)

                CNPJ  \
0     13255232000132   
1     71064976000176   
2     05200212000136   
3     15545768000118   
4     07561035000149   
...              ...   
1274  10850388000173   
1275  10530609000126   
1276  05705786000166   
1277  39395132000103   
1278  07857451000199   

                                                                       ADE  \
0        Ato Declaratório Executivo DRF/MCR nº 1, de 29 de janeiro de 2019   
1        Ato Declaratório Executivo DRF/MCR nº 1, de 29 de janeiro de 2019   
2      Ato Declaratório Executivo DRF/MCR nº 39, de 11 de dezembro de 2018   
3      Ato Declaratório Executivo DRF/MCR nº 37, de 20 de novembro de 2018   
4       Ato Declaratório Executivo DRF/MCR nº 31, de 25 de outubro de 2018   
...                                                                    ...   
1274        Ato Declaratório Executivo DRF/AJU nº 5, de 16 de maio de 2017   
1275        Ato Declaratório Executivo DRF/AJU nº 6, de 15 de maio de 2017   
1276     

C:\ProgramData\Anaconda3\lib\site-packages\pandas\core\generic.py:5208: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


Vamos validar os CNPJS:

In [75]:
# Função de validar CNPJ, copiada de https://github.com/rafahlobo/cpfValidator/blob/master/cpf.py
def isCnpjValid(cnpj):
    """ If cnpf in the Brazilian format is valid, it returns True, otherwise, it returns False. """

    # Check if type is str
    if not isinstance(cnpj,str):
        return False

    # Remove some unwanted characters
    cpf = re.sub("[^0-9]",'',cnpj)

    # Checks if string has 11 characters
    if len(cpf) != 14:
        return False

    sum = 0
    weight = [5,4,3,2,9,8,7,6,5,4,3,2]

    """ Calculating the first cpf check digit. """
    for n in range(12):
        value =  int(cpf[n]) * weight[n]
        sum = sum + value


    verifyingDigit = sum % 11

    if verifyingDigit < 2 :
        firstVerifyingDigit = 0
    else:
        firstVerifyingDigit = 11 - verifyingDigit

    """ Calculating the second check digit of cpf. """
    sum = 0
    weight = [6,5,4,3,2,9,8,7,6,5,4,3,2]
    for n in range(13):
        sum = sum + int(cpf[n]) * weight[n]

    verifyingDigit = sum % 11

    if verifyingDigit < 2 :
        secondVerifyingDigit = 0
    else:
        secondVerifyingDigit = 11 - verifyingDigit

    if cpf[-2:] == "%s%s" % (firstVerifyingDigit,secondVerifyingDigit):
        return True
    return False

In [83]:
# Remove os CNPJs inválidos:
empresas_baixadas_df = empresas_baixadas_df.drop(empresas_baixadas_df[empresas_baixadas_df.CNPJ.apply(isCnpjValid) == False].index)
empresas_baixadas_df

CNPJ  \
0     13255232000132   
1     71064976000176   
2     05200212000136   
3     15545768000118   
4     07561035000149   
...              ...   
1274  10850388000173   
1275  10530609000126   
1276  05705786000166   
1277  39395132000103   
1278  07857451000199   

                                                                       ADE  \
0        Ato Declaratório Executivo DRF/MCR nº 1, de 29 de janeiro de 2019   
1        Ato Declaratório Executivo DRF/MCR nº 1, de 29 de janeiro de 2019   
2      Ato Declaratório Executivo DRF/MCR nº 39, de 11 de dezembro de 2018   
3      Ato Declaratório Executivo DRF/MCR nº 37, de 20 de novembro de 2018   
4       Ato Declaratório Executivo DRF/MCR nº 31, de 25 de outubro de 2018   
...                                                                    ...   
1274        Ato Declaratório Executivo DRF/AJU nº 5, de 16 de maio de 2017   
1275        Ato Declaratório Executivo DRF/AJU nº 6, de 15 de maio de 2017   
1276       Ato Declaratório Executivo DRF/AJU nº 1, de 27 de março de 2017   
1277  Ato Declaratório Executivo DRF/VIT nº 129, de 19 de dezembro de 2019   
1278     Ato Declaratório Executivo DRF/PAL nº 17, de 23 de agosto de 2018   

                                                                                              Link  
0      http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98462  
1      http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=98462  
2      http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=97332  
3      http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96744  
4      http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=96270  
...                                                                                            ...  
1274   http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82850  
1275   http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=82849  
1276   http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=81572  
1277  http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=105647  
1278   http://normas.receita.fazenda.gov.br/sijut2consulta/link.action?visao=compilado&idAto=94367  

[1089 rows x 3 columns]

Salvando o resultdo com a lista de CNPJs de empresas que foram baixadas pela RFB, obtidas da pesquisa no site de normas:

In [84]:
empresas_baixadas_df.to_csv(r'C:\DATASETS\Empresas Baixadas\empresas_baixadas_extraido_site_normas.csv', index = False)